### PR TITLE
Fix expiring dict while flushing cache in scheduler

### PR DESCRIPTION
**Pull request by jpbruinsslot**
### Changes

It seems when the cache needs to be flushed, the time it takes to fill the cache the ttl could potentially pass and subsequently trigger an ExpiredError again which it then cannot handle anymore. With this change it would allow you to turn off the raising of an ExpiredError when you're trying to rebuild the cache. With this PR a additional boolean flag is added to an expiring dict, to signifiy to not raise an `ExpiredError` when using the the `ExpiringDict`.

### CODE REVIEW

**Comment by ammar92** [comment]
It's not immediately clear why you need the toggle, could you explain it a bit?

**Comment by underdarknl** [comment]
Is a more specified value not a more useful signature for this method?
Eg, set to True or False as an argument to this method, so as to not get in a position where everything gets inverted due to one missed call somewhere?

**Comment by jpbruinsslot** [comment]
> Is a more specified value not a more useful signature for this method? Eg, set to True or False as an argument to this method, so as to not get in a position where everything gets inverted due to one missed call somewhere?

Sure

**Comment by jpbruinsslot** [comment]
> It's not immediately clear why you need the toggle, could you explain it a bit?

It seems when the cache needs to be flushed, the time it takes to fill the cache the ttl could potentially pass and subsequently trigger an `ExpiredError` again which it then cannot handle anymore. With this change it would allow you to turn off the raising of an `ExpiredError` when you're trying to rebuild the cache.

File: mula/scheduler/utils/dict_utils.py [hunk 1]
```diff
@@ -37,12 +38,26 @@ def reset(self) -> None:
             self.cache.clear()
             self.expiration_time = datetime.now(timezone.utc) + self.lifetime
 
+    def set_expiration_enabled(self, value: bool) -> None:
```
Comment (ammar92) [inline mula/scheduler/utils/dict_utils.py:41]: This could also have been a property (e.g. using `@property.setter`)

**Review by ammar92 — COMMENTED** [review]
Looks good to me. I only had one remark that could be considered

**Comment by Darwinkel** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* No aparrent regressions
* No `ExpiredError` present in the scheduler logs

**Review by ammar92 — APPROVED** [review]

**Review by dekkers — APPROVED** [review]

**Pull request merged**